### DISCUSSION TITLE
Ruby Gems

### DISCUSSION THREAD

**Discussion by Nat-Spo** [usnistgov/macos_security#407] [General] [answered]
Hi there,

I am trying to run the application here on my MacBook using this guide https://golbiga.medium.com/getting-to-know-macos-security-compliance-project-part-1-31d0689b7c30 and I'm trying to run the ruby gems command located at the bottom of the setup scripts and It doesn't seem to work. Ive also tried to run the gem file command in the GitHub and tried to install the applications/packages manually but it doesn't seem to work. Any ideas? 
<img width="1056" alt="Screenshot 2024-07-18 at 12 13 13" src="https://github.com/user-attachments/assets/f2aa3e57-e41a-4ec9-8d54-86b041141295">
<img width="1280" alt="Screenshot 2024-07-18 at 12 14 02" src="https://github.com/user-attachments/assets/999a9f07-ee43-43a3-a52b-8b6adaf96dee">

**Comment by golbiga** [comment, marked as answer]
Looks like your network might be blocking on your network. That command worked just fine for me.

```
 bundle install --binstubs --path mscp_gems
Fetching gem metadata from https://rubygems.org/..........
Resolving dependencies...
Fetching Ascii85 1.1.1
Installing Ascii85 1.1.1
Fetching public_suffix 5.1.1
Installing public_suffix 5.1.1
Fetching addressable 2.8.7
Installing addressable 2.8.7
Fetching afm 0.2.2
Installing afm 0.2.2
Fetching asciidoctor 2.0.22
Installing asciidoctor 2.0.22
Fetching concurrent-ruby 1.3.3
Installing concurrent-ruby 1.3.3
Fetching matrix 0.4.2
Installing matrix 0.4.2
Fetching pdf-core 0.9.0
Installing pdf-core 0.9.0
Fetching ttfunk 1.7.0
Installing ttfunk 1.7.0
Fetching prawn 2.4.0
Installing prawn 2.4.0
Fetching prawn-icon 3.0.0
Installing prawn-icon 3.0.0
Fetching css_parser 1.12.0
Installing css_parser 1.12.0
Fetching rexml 3.2.6
Installing rexml 3.2.6
Fetching prawn-svg 0.34.2
Installing prawn-svg 0.34.2
Fetching prawn-table 0.2.2
Installing prawn-table 0.2.2
Fetching hashery 2.1.2
Installing hashery 2.1.2
Fetching ruby-rc4 0.1.5
Installing ruby-rc4 0.1.5
Fetching pdf-reader 2.12.0
Installing pdf-reader 2.12.0
Fetching prawn-templates 0.1.2
Installing prawn-templates 0.1.2
Fetching polyglot 0.3.5
Installing polyglot 0.3.5
Fetching treetop 1.6.12
Installing treetop 1.6.12
Fetching asciidoctor-pdf 2.3.17
Installing asciidoctor-pdf 2.3.17
Using bundler 1.17.2
Fetching rouge 3.30.0
Installing rouge 3.30.0
Bundle complete! 4 Gemfile dependencies, 24 gems now installed.
Bundled gems are installed into `./mscp_gems`
```

**Reply by Nat-Spo** [reply]
thanks for the update. I managed to get it running, I had to disable ipv6. I found the solution on stack overflow after a little bit of googling https://stackoverflow.com/questions/23659119/timeout-when-installing-ruby-gems. I also had to use sudo and admin privileges to write to the folder it tries to install it on, but this may be due to my MDM profiles and system configs rather than your program. 

Thanks again for your reply, and I have no solved the issue.

**Reply by golbiga** [reply]
You shouldn't use sudo as the gems should be installed locally and in the project folder. Where are you cloning the project?

**Reply by Nat-Spo** [reply]
it didn't give me permission to write to the folder where it tries to install it, so I had to use sudo. potentially the folder is installed in the wrong place, but I followed the steps in the guide I posted and this is what I had to do. /Users/nat******ler/macOS_security is the install path of the folder
